eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -29,10 +29,21 @@ import scala.collection.mutable
 object Graph {
 
   // @formatter:off
-  // A compound weight for an edge, weight is obtained with (cost X factor),'cost' contains the actual amount+fees in millisatoshi, 'cltvCumulative' the total CLTV necessary to reach this edge
+  /**
+   * The cumulative weight of a set of edges (path in the graph).
+   *
+   * @param cost   amount to send to the recipient + each edge's fees
+   * @param length number of edges in the path
+   * @param cltv   sum of each edge's cltv
+   * @param weight cost multiplied by a factor based on heuristics (see [[WeightRatios]]).
+   */
   case class RichWeight(cost: MilliSatoshi, length: Int, cltv: CltvExpiryDelta, weight: Double) extends Ordered[RichWeight] {
-    override def compare(that:  RichWeight): Int = this.weight.compareTo(that.weight)
+    override def compare(that: RichWeight): Int = this.weight.compareTo(that.weight)
   }
+  /**
+   * We use heuristics to calculate the weight of an edge based on channel age, cltv delta and capacity.
+   * We favor older channels, with bigger capacity and small cltv delta.
+   */
   case class WeightRatios(cltvDeltaFactor: Double, ageFactor: Double, capacityFactor: Double) {
     require(0 < cltvDeltaFactor + ageFactor + capacityFactor && cltvDeltaFactor + ageFactor + capacityFactor <= 1, "The sum of heuristics ratios must be between 0 and 1 (included)")
```

Comment:
It's currently possible to set values such as `capacityFactor=1234.5` and `ageFactor=-1234.4` which summed would lie within the interval [0, 1] but effectively screw up the cost function. How about adding a check to enforce individual correctness of the parameters?